pkg/controller/platformadmin/platformadmin_controller.go

```diff
@@ -600,3 +611,111 @@ func annotationToComponent(annotation map[string]string) ([]*config.Component, e
 
 	return components, nil
 }
+
+func NewYurtIoTCarrierComponent(platformAdmin *iotv1alpha2.PlatformAdmin) (*config.Component, error) {
```

Comment:
The build method can be considered as an internal function, and the build Deployment process can be placed in the util package.

Reply:
This function is still needed, but keep it as simple as possible.